src/lib/sql/sql_pipeline.cpp

```diff
@@ -1,163 +1,158 @@
+#include "sql_pipeline.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+#include <iomanip>
 #include <utility>
 
 #include "SQLParser.h"
 #include "concurrency/transaction_manager.hpp"
 #include "logical_query_plan/lqp_translator.hpp"
 #include "optimizer/optimizer.hpp"
 #include "scheduler/current_scheduler.hpp"
-#include "sql_pipeline.hpp"
-#include "sql_translator.hpp"
+#include "sql/sql_translator.hpp"
 #include "utils/assert.hpp"
 
 namespace opossum {
 
-SQLPipeline::SQLPipeline(const std::string& sql, std::shared_ptr<TransactionContext> transaction_context)
-    : _sql_string(sql), _use_mvcc(true), _auto_commit(false), _transaction_context(std::move(transaction_context)) {
-  DebugAssert(_transaction_context != nullptr, "Cannot pass nullptr as explicit transaction context.");
+std::vector<SQLPipeline> SQLPipeline::from_sql_string(const std::string& sql, const bool use_mvcc) {
+  // Pass no explicit transaction context so the SQLPipeline can decide whether it should create one or not.
+  return _from_sql_string(sql, nullptr, use_mvcc);
 }
 
-SQLPipeline::SQLPipeline(const std::string& sql, const bool use_mvcc)
-    : _sql_string(sql), _use_mvcc(use_mvcc), _auto_commit(_use_mvcc) {
-  if (_use_mvcc) {
-    // We want to use MVCC but didn't pass an explicit context
-    _transaction_context = TransactionManager::get().new_transaction_context();
-  } else {
-    // We don't want to use MVCC
-    _transaction_context = nullptr;
-  }
+std::vector<SQLPipeline> SQLPipeline::from_sql_string(const std::string& sql,
+                                                      std::shared_ptr<TransactionContext> transaction_context) {
+  DebugAssert(transaction_context != nullptr, "Cannot pass nullptr as explicit transaction context.");
+  return _from_sql_string(sql, transaction_context, true);
 }
 
-const hsql::SQLParserResult& SQLPipeline::get_parsed_sql() {
-  if (_parsed_sql) {
-    // Returned cached result
-    return *(_parsed_sql.get());
-  }
-
-  auto parse_result = std::make_unique<hsql::SQLParserResult>();
-
-  const auto started = std::chrono::high_resolution_clock::now();
+std::vector<SQLPipeline> SQLPipeline::_from_sql_string(const std::string& sql,
+                                                       std::shared_ptr<TransactionContext> transaction_context,
+                                                       bool use_mvcc) {
```

Comment:
Do we need the context parameter here? If we pass a context, we shouldn't have to pass use_mvcc. If we set use_mvcc, passing a context makes no sense. This is only for the case that we do not have an explicit context and want to decide if an implicit context should be created or not, correct?